tests/unit-tests/dds/DCPS/GuidGenerator.cpp

```diff
@@ -43,30 +43,37 @@ TEST(dds_DCPS_GuidGenerator, maintest)
     gen.populate(g2);
     EXPECT_TRUE(not_null(g1));
     EXPECT_TRUE(compare_prefix(g1,g2) != 0);
-  }
+
+
+    int val1 = gen.interfaceName("Test_NotARealInterface");
+    int val2 = gen.interfaceName("Test_NotARealInterface"); // synthesize a shortcut "found" that is a false success
+
+    EXPECT_TRUE(val1 == -1); // This is a valid failure
+    EXPECT_TRUE(val2 == 0); // but methinks the shortcut is a bug
+}
 
   // Test GUID converter
   {
     GUID_t guid;
-    guid.guidPrefix[ 0] =  1;
```

Comment:
This test block for `GuidConverter` doesn't belong here.  Please move it `tests/unit-tests/dds/DCPS/GuidConverter.cpp`.  Name it according to the guidlines in https://github.com/objectcomputing/OpenDDS/blob/master/docs/internal/unit_tests.rst